src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -226,6 +227,55 @@ public function hydrate(object $component, array $props, array $updatedProps, Li
         return $attributes;
     }
 
+    /**
+     * Hydrate a value from a dehydrated value.
+     *
+     * Depending on the prop configuration, the value may be hydrated by a custom method or the Serializer component.
+     *
+     * @throws SerializerExceptionInterface
+     */
+    public function hydrateValue(mixed $value, LivePropMetadata $propMetadata, object $parentObject): mixed
+    {
+        if ($propMetadata->hydrateMethod()) {
+            if (!method_exists($parentObject, $propMetadata->hydrateMethod())) {
+                throw new \LogicException(sprintf('The "%s" object has a hydrateMethod of "%s" but the method does not exist.', $parentObject::class, $propMetadata->hydrateMethod()));
+            }
+
+            return $parentObject->{$propMetadata->hydrateMethod()}($value);
+        }
+
+        if ($propMetadata->useSerializerForHydration()) {
+            return $this->normalizer->denormalize($value, $propMetadata->getType(), 'json', $propMetadata->serializationContext());
```

Comment:
$propMetadata->getType is nullable

Reply:
Should we throw an exception if `$propMetadata->getType()` is `null`? Like

```php
if (null === $propMetadata->getType()) {
    throw new \LogicException(sprintf('The "%s::%s" object should be hydrated with the Serializer, but no type could be guessed.', $parentObject::class, $propMetadata->getName()));
}
```

Reply:
Maybe yes ! I don't have a good vision on the impacts, but i don't see many cases where we'd be happy with un undefined type